File: core/ValidPalindrome/ValidPalindrome.py
def isPalindrome(s: str) -> bool:
    left, right = 0, len(s) - 1
    s = s.lower()
    while left < right:
        while left < right and not s[left].isdigit() and not s[left].isalpha():
            left += 1
        while left < right and not s[right].isdigit() and not s[right].isalpha():
            right -= 1
        if left < right and s[left] != s[right]:
            return False
        left += 1
        right -= 1
    return True

File: core/ValidPalindrome/ValidPalindrome.java
package com.dbc.code;

import java.util.Locale;

public class ValidPalindrome {
    public boolean isPalindrome(String s) {
        int left = 0, right = s.length() - 1;
        s = s.toLowerCase();
        while (left < right) {
            while (left < right && !Character.isDigit(s.charAt(left)) && !Character.isLetter(s.charAt(left))) left++;
            while (left < right && !Character.isDigit(s.charAt(right)) && !Character.isLetter(s.charAt(right))) right--;
            if (left < right && s.charAt(left) != s.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}
